src/PaddedViews.jl

```diff
@@ -186,6 +186,11 @@ Base.@propagate_inbounds function Base.setindex!(A::PaddedView{T, N}, v, i::Vara
     return A
 end
 
+Base.@propagate_inbounds function Base.setindex!(A::PaddedView, v, i::Int)
+    # delay the boundscheck in the IndexCartesian version and not set @inbounds meta here
+    setindex!(A, v, Base._to_subscript_indices(A, i...)...)
+end
+
```

Comment:
The weird thing here is that even if this is not added, the test still passes with `pkg> test` but not with `pkg> include("test/runtest.jl")`. Either I'm using `@boundscheck` wrongly or there's some bug upstream...